backend/typescript/services/implementations/camperService.ts

```diff
@@ -571,56 +687,114 @@ class CamperService implements ICamperService {
     }
   }
 
-  async deleteCamperById(camperId: string): Promise<void> {
+  async deleteCampersById(camperIds: Array<string>): Promise<void> {
     try {
-      const camperToDelete: Camper | null = await MgCamper.findById(camperId);
+      const campers: Array<Camper> | null = await MgCamper.find({
+        _id: {
+          $in: camperIds,
+        },
+      });
 
-      if (!camperToDelete) {
-        throw new Error(`Camper with camper ID ${camperId} not found.`);
+      if (campers === null || campers.length !== camperIds.length) {
+        throw new Error(
+          `Not all campers with camper IDs [${camperIds}] are found.`,
+        );
+      }
+
+      const oneChargeId = campers[0].chargeId;
+      for (let i = 0; i < campers.length; i += 1) {
+        if (campers[i].chargeId !== oneChargeId) {
+          throw new Error(`ChargeIds must all be the same.`);
+        }
       }
 
       const campSession: CampSession | null = await MgCampSession.findById(
-        camperToDelete.campSession,
+        campers[0].campSession,
       );
       if (!campSession) {
         throw new Error(
-          `Camper's camp session with campId ${camperToDelete.campSession} not found.`,
+          `Camp session with ID ${campers[0].campSession} not found.`,
         );
       }
 
+      /* eslint-disable no-await-in-loop */
+      for (let i = 0; i < campers.length; i += 1) {
+        const camperCampSession: CampSession | null = await MgCampSession.findById(
+          campers[i].campSession,
+        );
+        if (!camperCampSession) {
+          throw new Error(
+            `Campers' camp session with campSessionId ${campers[i].campSession} not found.`,
+          );
+        }
+        if (camperCampSession.id.toString() !== campSession.id.toString()) {
+          throw new Error(
+            `Not all campers are registered for the same campSession.`,
+          );
+        }
+      }
```

Comment:
like my other comment above we have `campSessionId` in campers so we can do this check without retrieving each campSession